src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
```

Comment:
I don't think the FS layer should return a token stream, but we could have a separate plumping in between the two.

What do you think of the following:

`LiquidRendering -> TemplateFactory -> TemplateRepository`

If you like this structure, then the `LiquidOptions` would take a `TemplateFactory`, that would return something higher-level than a string; probably a template.